storagenode/contact/service.go

```diff
@@ -0,0 +1,60 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package contact
+
+import (
+	"sync"
+	"time"
+
+	"go.uber.org/zap"
+	"gopkg.in/spacemonkeygo/monkit.v2"
+
+	"storj.io/storj/pkg/storj"
+	"storj.io/storj/pkg/transport"
+	"storj.io/storj/satellite/overlay"
+)
+
+var mon = monkit.Package()
+
+// Service is the contact service between storage nodes and satellites
+type Service struct {
+	log       *zap.Logger
+	self      *overlay.NodeDossier
+	transport transport.Client
+
+	mu               sync.Mutex
+	lastPinged       time.Time
+	whoPingedNodeID  storj.NodeID
+	whoPingedAddress string
```

Comment:
I wonder whether we should keep this information per source?

That way dashboard could show status per satellite.

Reply:
That would be good, but I think it would be far lower in priority than finishing kademlia removal.